sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/InstrumentBuilder.java

```diff
@@ -15,95 +16,72 @@
 import io.opentelemetry.sdk.metrics.internal.state.SdkObservableMeasurement;
 import io.opentelemetry.sdk.metrics.internal.state.WriteableMetricStorage;
 import java.util.Collections;
+import java.util.List;
 import java.util.function.BiFunction;
 import java.util.function.Consumer;
+import javax.annotation.Nullable;
 
 /** Helper to make implementing builders easier. */
-abstract class AbstractInstrumentBuilder<BuilderT extends AbstractInstrumentBuilder<?>> {
-
-  static final String DEFAULT_UNIT = "";
+final class InstrumentBuilder {
 
+  private final String name;
   private final MeterProviderSharedState meterProviderSharedState;
-  private final InstrumentType type;
-  private final InstrumentValueType valueType;
-  private String description;
-  private String unit;
-
-  protected final MeterSharedState meterSharedState;
-  protected final String instrumentName;
-  protected final Advice.AdviceBuilder adviceBuilder;
-
-  AbstractInstrumentBuilder(
-      MeterProviderSharedState meterProviderSharedState,
-      MeterSharedState meterSharedState,
-      InstrumentType type,
-      InstrumentValueType valueType,
+  private final MeterSharedState meterSharedState;
+  @Nullable private InstrumentType type = null;
+  @Nullable private InstrumentValueType valueType;
```

Comment:
type and valueType should be construction parameters since they should never be null, right?

Reply:
Ugh, well yeah, that's the other approach. The nullability and validation didn't feel great.

The problem will then be that the builder constructor will take 5 arguments. This is something that the `MutableInstrumentDescriptor` helped with -- reducing the constructor arg count.

I'll just eat the argument cost and see what it looks like. Since it's package-private buildery stuff anyway, I probably won't lose sleep over it.

Reply:
A builder that has more constructor args than settable fields feels like a smell to me.

Reply:
Yeah its a bit of a smell, but so is having nullable fields which are actually required 😛.

Reply:
Truth. 🙃